metldata/builtin_transformations/aggregate/model_transform.py

```diff
@@ -0,0 +1,370 @@
+    name and range is inferred from the original input classes.
+
+    Args:
+        input_model (MetadataModel): _description_
+        output_model (MetadataModel): _description_
+        class_map (list[tuple[str, str]]): _description_
+
+    Raises:
+        MetadataModelTransformationError: _description_
+
+    Returns:
+        MetadataModel: _description_
+    """
+    output_schema_view = output_model.schema_view
+    for in_class_name, out_class_name in origin_map.items():
+        slot_name = id_slot_map[out_class_name]
+        input_slot = input_model.schema_view.induced_slot(slot_name, in_class_name)
+        output_schema_view = upsert_class_slot(
+            schema_view=output_schema_view,
+            class_name=out_class_name,
+            new_slot=SlotDefinition(
+                name=slot_name, range=input_slot.range, identifier=True
+            ),
+        )
+
+    return output_schema_view.export_model()
+
+
+def build_aggregation_model(
+    model: MetadataModel, config: AggregateConfig
+) -> MetadataModel:
+    """Create a data model for the result of an aggregation transformation as
+    described in an AggregateConfig
+
+    Args:
+        model (MetadataModel): The LinkML model to transform
+        config (AggregateConfig): The config
+
+    Returns:
+        MetadataModel: A new, modified MetadataModel
+    """
+    # Reduce model to a bare bone model
+
+    min_output_model = MinimalLinkMLModel()
+
+    # Add function output classes
+    for agg_func in (
+        op.function for agg in config.aggregations for op in agg.operations
+    ):
+        cls_def = agg_func.result_range_cls_def()
+        cls_name = agg_func.result_range_name()
+        if cls_def:
+            min_output_model.add_named_class(cls_def=cls_def, cls_name=cls_name)
+
+    # Add upstream classes
+    for aggregation in config.aggregations:
+        add_aggregation(min_model=min_output_model, aggregation=aggregation)
+
+    output_model = bare_bone_model_from_other(model)
+    # Add slots and classes to the bare bone metadata model
+    update_metadata_model(output_model, min_output_model)
+
+    # Add identifier slots to classes
+    origin_map = {agg.input: agg.output for agg in config.aggregations}
+    id_slot_map = get_identifier_slot_names(input_model=model, origin_map=origin_map)
+    output_model = add_identifier_slots(
+        input_model=model,
+        output_model=output_model,
+        origin_map=origin_map,
+        id_slot_map=id_slot_map,
+    )
+
+    # Anchor the output classes
+    schema_view = output_model.schema_view
+    for aggregation in config.aggregations:
+        schema_view = add_anchor_point(
+            schema_view=schema_view,
+            anchor_point=AnchorPoint(
+                target_class=aggregation.output,
+                identifier_slot=id_slot_map[aggregation.output],
+                root_slot=snakecase(aggregation.output) + "s",
```

Comment:
This will create a slot "dataset_statss"...

Reply:
Yes, I know. Do you have a nicer rule in mind? 😅 Shall we check for a terminating `s`? 🤷

Reply:
That should be good enough. Until we have a class called "Virus".